PocketKit/Sources/PocketKit/Home/SlateDetailViewController.swift

```diff
@@ -186,52 +193,131 @@ private extension SlateDetailViewController {
             return NSCollectionLayoutSection(group: group)
         case .slate(let slate):
             let width = environment.container.effectiveContentSize.width
-            let margin: CGFloat = environment.traitCollection.shouldUseWideLayout() ? Margins.iPadNormal.rawValue : Margins.normal.rawValue
-
+            let sideMargin: CGFloat = environment.traitCollection.shouldUseWideLayout() ? Margins.iPadNormal.rawValue : Margins.normal.rawValue
             let recommendations = slate.recommendations?.compactMap { $0 as? Recommendation } ?? []
 
-            let components = recommendations.reduce((CGFloat(0), [NSCollectionLayoutItem]())) { result, recommendation in
-                guard let viewModel = self.model.recommendationViewModel(for: recommendation.objectID) else {
-                    return result
-                }
-
-                let currentHeight = result.0
-                let height = RecommendationCell.fullHeight(viewModel: viewModel, availableWidth: width - (margin * 2)) + margin
-                var items = result.1
-                let item = NSCollectionLayoutItem(
-                    layoutSize: NSCollectionLayoutSize(
-                        widthDimension: .fractionalWidth(1),
-                        heightDimension: .absolute(height)
-                    )
-                )
-                item.contentInsets = NSDirectionalEdgeInsets(top: 8, leading: 0, bottom: 8, trailing: 0)
+            if environment.traitCollection.shouldUseWideLayout() {
+                return slateSectionForWideLayout(with: recommendations, width: width, sideMargin: sideMargin)
+            } else {
+                return slateSectionForCompact(with: recommendations, width: width, sideMargin: sideMargin)
+            }
+        default:
+            return .empty()
+        }
+    }
 
-                items.append(item)
+    /// Determines the section layout for the slate detail view on iPad mode with regular horizontal size class (including split view). Number of columns for the grid layout depends on device orientation and the view's width length. If view's width is less than 800, show 2 col, otherwise show 2 col if it is in portrait and 3 col if it is in landscape mode.
+    private func slateSectionForWideLayout(with recommendations: [Recommendation], width: CGFloat, sideMargin: CGFloat) -> NSCollectionLayoutSection {
+        if self.view.bounds.width >= Constants.minWidthBoundaryForFullColumnLayout && !UIDevice.current.orientation.isPortrait {
```

Comment:
The condition is hard to parse, would it be possible to pull into something like `func viewWidthExceedsPortraitMinimum() -> Bool`

Reply:
I updated it to a private computed variable 
```
    private var showThreeColumnLayout: Bool {
        return self.view.bounds.width >= Constants.minWidthBoundaryForFullColumnLayout && !UIDevice.current.orientation.isPortrait
    }
```

Reply:
^ outdated, I made some refactoring changes based on a combination of yours and Giorgio's comments.